frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -27,3 +42,89 @@ frappe.ui.form.on("System Settings", "enable_two_factor_auth", function(frm) {
 		frm.set_value("bypass_restrict_ip_check_if_2fa_enabled", 0);
 	}
 });
+
+frappe.ui.form.on("System Settings", "enable_backup_encryption", function (frm) {
+    if ((frm.doc.enable_backup_encryption && frm.doc.new_password_encryption != frm.doc.confirm_password_encryption) || !frm.doc.confirm_password_encryption) {
+
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+
+    } else {
+        frm.doc.new_password_encryption = "";
+        frm.doc.confirm_password_encryption = "";
+        frm.doc.password_encryption = "";
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
+
+        } else {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: red"
+
+        }
+
+    }
+
+});
+
+frappe.ui.form.on("System Settings", "change_encryption_password", function (frm) {
+    if (frm.doc.enable_backup_encryption && frm.doc.change_encryption_password) {
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.new_password_encryption.input_area.children[0].value = "******";
+        cur_frm.fields_dict.confirm_password_encryption.input_area.children[0].value = "******";
+
+
+    } else {
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.fields_dict.new_password_encryption.df.hidden = 1;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 1;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
```

Comment:
Can we make this less yucky? https://frappe.io/docs/user/en/tutorial/form-client-scripting

Reply:
thanks for pointing this out, we're now using frappe.ui.form.on method, removed it inside the conditions and initialized it outside to avoid redundancy.